components/centraldashboard/dashboard.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"regexp"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	// Uncomment the following line to load the gcp plugin (only required to authenticate against GKE clusters).
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+var (
+	port = os.Getenv("PORT_1")
+)
+
+// Call GetClient from client/manager.go to obtain Cluster configs.
+// Use client info from GetClient to call SearchKatib from findservice/findservice.go.
+// SearchKatib will return a boolean true if Katib is deployed.
+
+func main() {
+
+	clientset := GetClient()
+	isKatibDeployed := SearchKatibPods(clientset)
+
+	indexServer := http.FileServer(http.Dir("frontend/layout/"))
+	if isKatibDeployed {
+		indexServer = http.FileServer(http.Dir("frontend/index/"))
+	}
+
+	http.Handle("/", indexServer)
+	log.Println("Listening on", ":"+port)
+	log.Fatal(http.ListenAndServe(":"+port, nil))
+}
+
+// GetClient gets client information
+func GetClient() *kubernetes.Clientset {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		panic(err.Error())
+	}
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	return clientset
+}
+
+//SearchKatibPods searches across the cluster to see if Katib is deployed.
+func SearchKatibPods(clientset *kubernetes.Clientset) bool {
+
+	pods, err := clientset.CoreV1().Pods("").List(metav1.ListOptions{})
+	if err != nil {
+		fmt.Printf("Error in regexp")
+	}
+
+	katibpod := "vizier*"
```

Comment:
You don't understand the regexp syntax and matching. "*" denotes a zero or more occurrences. It matches on "vizie". It must be "^vizier", or better use strings.HasPrefix.

 func example() {
    r, _ := regexp.Compile("foo*")

    fmt.Println(r.MatchString("food"))
    fmt.Println(r.MatchString("seafood"))
    fmt.Println(r.MatchString("info"))
    fmt.Println(strings.HasPrefix("food", "foo"))
    fmt.Println(strings.HasPrefix("seafood", "foo"))
}